src/mudlet.cpp

```diff
@@ -2378,19 +2369,20 @@ QString mudlet::readProfileData( const QString& profile, const QString& item )
 // this slot is called via a timer in the constructor of mudlet::mudlet()
 void mudlet::startAutoLogin()
 {
-    QStringList hostList = QDir(QDir::homePath()+"/.config/mudlet/profiles").entryList(QDir::Dirs, QDir::Name);
-    hostList.removeAll(".");
-    hostList.removeAll("..");
-    for( int i = 0; i< hostList.size(); i++ )
-    {
-        QString item = "autologin";
-        QString val = readProfileData( hostList[i], item );
-        if( val.toInt() == Qt::Checked )
-        {
-            doAutoLogin( hostList[i] );
+    QStringList hostList = QDir(QDir::homePath() + "/.config/mudlet/profiles").entryList(QDir::Dirs | QDir::NoDotAndDotDot, QDir::Name);
+    bool openedProfile = false;
+
+    for (auto host : hostList) {
+        QString val = readProfileData(host, QStringLiteral("autologin"));
```

Comment:
This is one of those cases where I think we can/should use the `QLatin1String("autologin")` instead - as there will be at one more instance of this, once we check all the strings!

📖  You have reminded us that the `QStringLiteral(`...`)` form - though more convenient when a string might be translated after all and thus switch to a `tr(`...`)` construct and also supporting positional place-holder (`%1` to `%99`) replacements {whereas the `QLatin1String(`...`)` form does not} the `QLatin1String` usage does allow shared usage of the compiled constants whereas the `QStringLiteral` being storable in a read-only data segment does not...